### ISSUE TITLE
[Bug]  CasaOS fails to update on RBP 5

### ISSUE THREAD

**Issue by bpace** [IceWhaleTech/CasaOS#1664]
**Describe the bug**

CasaOS was failing to install on my RBP 5 so I searched for a solution. I found this: https://github.com/IceWhaleTech/CasaOS/issues/1490
They solved the issue by installing from this link:  curl -fsSL https://get.casaos.io/v0.4.4-3-rpi5 | sudo bash

Everything installed fine only now I can't update CasaOS as it fails to run the migration script.

```
Update
OK Your hardware architecture is : aarch64
INFO Configuration CasaOS Addons
INFO Downloading https://github.com/IceWhaleTech/CasaOS-Gateway/releases/download/v0.4.4-3-alpha5/linux-arm64-casaos-gateway-v0.4.4-3-alpha5.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS-MessageBus/releases/download/v0.4.4-3-alpha2/linux-arm64-casaos-message-bus-v0.4.4-3-alpha2.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS-UserService/releases/download/v0.4.5/linux-arm64-casaos-user-service-v0.4.5.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS-LocalStorage/releases/download/v0.4.4/linux-arm64-casaos-local-storage-v0.4.4.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS-AppManagement/releases/download/v0.4.5/linux-arm64-casaos-app-management-v0.4.5.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS/releases/download/v0.4.6/linux-arm64-casaos-v0.4.6.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS-CLI/releases/download/v0.4.4-3-alpha1/linux-arm64-casaos-cli-v0.4.4-3-alpha1.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS-UI/releases/download/v0.4.6/linux-all-casaos-v0.4.6.tar.gz...
INFO Downloading https://github.com/IceWhaleTech/CasaOS-AppStore/releases/download/v0.4.5/linux-all-appstore-v0.4.5.tar.gz...
INFO Extracting linux-all-appstore-v0.4.5.tar.gz...
INFO Extracting linux-all-casaos-v0.4.6.tar.gz...
INFO Extracting linux-arm64-casaos-app-management-v0.4.5.tar.gz...
INFO Extracting linux-arm64-casaos-cli-v0.4.4-3-alpha1.tar.gz...
INFO Extracting linux-arm64-casaos-gateway-v0.4.4-3-alpha5.tar.gz...
INFO Extracting linux-arm64-casaos-local-storage-v0.4.4.tar.gz...
INFO Extracting linux-arm64-casaos-message-bus-v0.4.4-3-alpha2.tar.gz...
INFO Extracting linux-arm64-casaos-user-service-v0.4.5.tar.gz...
INFO Extracting linux-arm64-casaos-v0.4.6.tar.gz...
INFO Running /tmp/casaos-installer/tmp.FSmh0bscpJ/build/scripts/migration/script.d/01-migrate-gateway.sh...
INFO Running /tmp/casaos-installer/tmp.FSmh0bscpJ/build/scripts/migration/script.d/02-migrate-user-service.sh...
INFO Running /tmp/casaos-installer/tmp.FSmh0bscpJ/build/scripts/migration/script.d/03-migrate-casaos.sh...
INFO Running /tmp/casaos-installer/tmp.FSmh0bscpJ/build/scripts/migration/script.d/04-migrate-local-storage.sh...
FAILED Failed to run migration script
```

**To Reproduce**

1. Install Raspberry PI OS Lite (64-bit)
2. on the Terminal use `curl -fsSL https://get.casaos.io/v0.4.4-3-rpi5 | sudo bash`
3. Once CasaOS is installed go to Settings > Update
4. See error FAILED "Failed to run migration script"

**Expected behavior**

I expected this to update CasaOS

**Desktop (please complete the following information):**

```
 - OS: Debain Bookworm, headless
 - Browser: using chrome on windows to remote in
 - Version: CasaOS v0.4.4-3
```

**System Time**

```
`Local time: Sun 2024-01-28 18:14:12 EST
           Universal time: Sun 2024-01-28 23:14:12 UTC
                 RTC time: Sun 2024-01-28 23:14:12
                Time zone: America/New_York (EST, -0500)
System clock synchronized: yes
              NTP service: active
          RTC in local TZ: no`
```

**Logs**
```
Jan 27 19:25:33 raspberrypi58 systemd[1]: Starting casaos-local-storage.service - CasaOS Local Storage Service...
░░ Subject: A start job for unit casaos-local-storage.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A start job for unit casaos-local-storage.service has begun execution.
░░
░░ The job identifier is 132.
Jan 27 19:25:34 raspberrypi58 casaos-local-storage[953]: v0.4.4
Jan 27 19:25:34 raspberrypi58 casaos-local-storage[990]: git commit: 4b68959866dd8b4a3d176f5f2447ab73ac1dd267
Jan 27 19:25:34 raspberrypi58 casaos-local-storage[990]: build date: 2023-11-20T09:41:11Z
Jan 27 19:25:34 raspberrypi58 casaos-local-storage[990]: 2024-01-27T19:25:34.693-0500        info        Checking serial disk mount...        {"func": "service.(*diskService).CheckSerialDiskMount", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/disk.go", "line": 552}
Jan 27 19:25:34 raspberrypi58 casaos-local-storage[990]: 2024-01-27T19:25:34.744-0500        info        Notified systemd that local storage service is ready        {"func": "main.main", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/main.go", "line": 204}
Jan 27 19:25:34 raspberrypi58 casaos-local-storage[990]: 2024-01-27T19:25:34.744-0500        info        LocalStorage service is listening...        {"address": "127.0.0.1:42679", "func": "main.main", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/main.go", "line": 209}
Jan 27 19:25:34 raspberrypi58 systemd[1]: Started casaos-local-storage.service - CasaOS Local Storage Service.
░░ Subject: A start job for unit casaos-local-storage.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A start job for unit casaos-local-storage.service has finished successfully.
░░
░░ The job identifier is 132.
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: smartctl exit status 1
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: smartctl /dev/mmcblk0
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: smartctl 593
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: smartctl exit status 1
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: smartctl /dev/mmcblk0
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: smartctl 593
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: 2024-01-27T19:25:35.116-0500        error        failed to enable S.M.A.R.T: smartctl 7.3 2022-02-28 r5338 [aarch64-linux-6.1.0-rpi7-rpi-2712] (local build)
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: /dev/mmcblk0: Unable to detect device type
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: Please specify device type with the -d option.
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]: Use smartctl -h to get a usage summary
Jan 27 19:25:35 raspberrypi58 casaos-local-storage[990]:         {"error": "exit status 1", "path": "/dev/mmcblk0", "func": "service.(*diskService).CheckSerialDiskMount", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/disk.go", "line": 576}
Jan 27 19:26:29 raspberrypi58 casaos-local-storage[990]: 2024-01-27T19:26:29.086-0500        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
Jan 28 17:06:52 raspberrypi58 casaos-local-storage[990]: 2024-01-28T17:06:52.773-0500        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
Jan 28 18:07:28 raspberrypi58 casaos-local-storage[990]: 2024-01-28T18:07:28.284-0500        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
Jan 28 18:19:05 raspberrypi58 casaos-local-storage[990]: 2024-01-28T18:19:05.466-0500        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
Jan 28 18:54:56 raspberrypi58 casaos-local-storage[990]: 2024-01-28T18:54:56.146-0500        error        failed to publish event to message bus        {"status": "404 Not Found", "responseError": "json: unsupported type: func() (io.ReadCloser, error)", "func": "service.(*notifyServer).SendNotify", "file": "/home/runner/work/CasaOS-LocalStorage/CasaOS-LocalStorage/service/notify.go", "line": 32}
```

**Comment by x3llaz** [comment]
I have similar "failed to run migration script" issue on CasaOS v0.4.5 when trying to update to 0.4.6. I also can't change my wallpaper anymore when using 0.4.5. I ended up just downgrading back to 0.4.3 and everything works again. I'm not even going to update anymore for now as 0.4.5 and 0.4.6 seems to be very buggy. 🫤

Here's the the thread I opened with similar issues to yours 5 days ago:
[#1647](https://github.com/IceWhaleTech/CasaOS/issues/1647)
